src/GlobalPayments.Api/Terminals/INGENICO/Responses/DeviceResponse.cs

```diff
@@ -0,0 +1,141 @@
+﻿using GlobalPayments.Api.Entities;
+using GlobalPayments.Api.Terminals.Abstractions;
+using GlobalPayments.Api.Terminals.Extensions;
+using System.IO;
+using System.Text;
+using GlobalPayments.Api.Terminals;
+using GlobalPayments.Api.Utils;
+using System;
+
+namespace GlobalPayments.Api.Terminals.Ingenico {
+    public abstract class IngenicoBaseResponse : IDeviceResponse {
+        public string Status { get; set; }
+        public string Command { get; set; }
+        public string Version { get; set; }
+        public string DeviceResponseCode { get; set; }
+        public string DeviceResponseText { get; set; }
+        public string ReferenceNumber { get; set; }
+    }
+
+    public class IngenicoTerminalResponse : IngenicoBaseResponse, ITerminalResponse {
+
+        internal string _transactionStatus;
+        internal decimal _amount;
+        internal PaymentMode _paymentMode;
+        internal string _privateData;
+        internal string terminalRawData;
+        internal string _currencyCode;
+        internal DataResponse _respField;
+        
+        private byte[] _buffer;
+        private ParseFormat _parseFormat;
+
+        internal IngenicoTerminalResponse(byte[] buffer, ParseFormat format = ParseFormat.Transaction) {
+            _buffer = buffer;
+            _parseFormat = format;
+            ParseResponse(buffer);
+        }
+
+        public override string ToString() {
+            base.DeviceResponseText = Encoding.UTF8.GetString(_buffer, 0, _buffer.Length);
+            return base.DeviceResponseText;
+        }
+
+        #region Added Properties Specific for Ingenico
+        public string DccCurrency { get { return _respField.DccCode; } set { } }
+        public DynamicCurrencyStatus? DccStatus { get { return _respField.DccStatus; } set { } }
+        public decimal? DccAmount { get { return _respField.DccAmount; } set { } }
+        public TransactionSubTypes? TransactionSubType { get { return _respField.TransactionSubType; } set { } }
+        public decimal? SplitSaleAmount { get { return 0; } set { } }
+        public PaymentMode PaymentMode { get { return _paymentMode; } set { } }
+        public string CurrencyCode { get { return _currencyCode; } set { } }
+        public string PrivateData { get { return _privateData; } }
+        public decimal? FinalTransactionAmount { get { return _respField.FinalAmount; } }
+        #endregion
+
+        #region Properties
+        public string ResponseText { get { return terminalRawData; } set { } }
```

Comment:
This property is not the full raw response but the response text which accompanies the response code.

i.e. "00" response code = "Success" response text